drivers/eeprom/i2c_xx24xx.c

```diff
@@ -266,6 +267,7 @@ static int ee24xx_waitwritecomplete(FAR struct ee24xx_dev_s *eedev,
   do
     {
      ret = I2C_TRANSFER(eedev->i2c, msgs, 1);
+     up_udelay(10);
```

Comment:
up_udelay() should only be used during initialize.  It has disasterous consequences on really time performance since it is a tight loop and blocks all other tasks.  The code was better without this because you had  chance  to get out of the loop sooner.  No, there is almost a guaranteed block, hogging the CPU for a minimum of 10 usec.

You got some very bad advice.